### PR TITLE
SSR-2247 user guide and demo app typescript

### PR DESCRIPTION
**Pull request by voduchau**

### CODE REVIEW

File: demos/typescript-app/dist/index.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <title>Typescript demo app</title>
```
Comment (TomokoMiyake) [inline demos/typescript-app/dist/index.html:5]: Change from Typescript > TypeScript

File: demos/typescript-app/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Dropdown, DropdownProps } from "kintone-ui-component";
```
Comment (TomokoMiyake) [inline demos/typescript-app/src/index.ts:1]: Unit test is failing now. Please check it.

File: demos/typescript-app/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      value: "-----",
+    },
+    {
+      label: "Orange",
+      value: "orange",
+    }
+  ],
+  value: "-----",
+  label: "フルーツ一覧",
+  error: "エラーです",
```
Comment (TomokoMiyake) [inline demos/typescript-app/src/index.ts:17]: Please change to English.

label: "Fruit",
error: "Error occurred!",

Please add one more items option as an example.

File: demos/typescript-app/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "typescript-app",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack --watch"
```
Comment (TomokoMiyake) [inline demos/typescript-app/package.json:8]: I think there is no test script and no need of this script.
About build, please unify with esm-demo-app way. Or do you think this setting is good for TS tryout?

File: demos/typescript-app/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+    "compilerOptions": {
+      "outDir": "./dist/",
+      "noImplicitAny": true,
+      "module": "es6",
+      "target": "es5",
+      "jsx": "react",
+      "allowJs": true,
+      "moduleResolution": "node"
```
Comment (TomokoMiyake) [inline demos/typescript-app/tsconfig.json:9]: Please refer to the root tsconfig setting and update it.
It seems there are some unnecessary parameters. (ex. jsx)

File: demos/typescript-app/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      },
+    ],
+  },
+  resolve: {
+    extensions: ['.tsx', '.ts', '.js'],
+  },
+  output: {
+    filename: 'bundle.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
```
Comment (TomokoMiyake) [inline demos/typescript-app/webpack.config.js:20]: Please refer to the root webpack.config setting.
It seems there are some unnecessary description. (ex. tsx)

I think it's ok with main.js as filename as same as esm-demo-app. How do you think?

**Review by TomokoMiyake — CHANGES_REQUESTED** [review]
Please merge the latest develop branch.
npm ci is failing now. I wanna check the behavior with npm link.

Please add README as a guide like esm-demo-app.

About bundle setting, I haven't reviewed in detail. So let's ask PG reviewer afterward.

File: demos/typescript-app/dist/bundle.js.LICENSE.txt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
```
[the remark this answers is not in the document]
Comment (voduchau) [reply demos/typescript-app/dist/bundle.js.LICENSE.txt:17]: we don't need it, I will remove it

File: demos/typescript-app/package.json (hunk 4 above)
Comment (voduchau) [reply demos/typescript-app/package.json:8]: I will unify with esm-demo-app

File: demos/typescript-app/tsconfig.json (hunk 5 above)
Comment (voduchau) [reply demos/typescript-app/tsconfig.json:9]: I will refer and update it

File: demos/typescript-app/webpack.config.js (hunk 6 above)
Comment (voduchau) [reply demos/typescript-app/webpack.config.js:20]: Yes, I will refer and update it

**Review by TomokoMiyake — CHANGES_REQUESTED** [review]
Thanks for the improvement!

There is no content in .gitignore. Will it be better to set node_modules, etc?

File: demos/typescript-app/package.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "build:dev": "webpack --watch --mode development",
+    "build:prod": "webpack --watch --mode production"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (voduchau) [reply demos/typescript-app/package.json:13]: >For managing source code, let's apply a lint package, too.
ok, I will apply a lint package.

File: demos/typescript-app/package.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "typescript-app",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "build:dev": "webpack --watch --mode development",
+    "build:prod": "webpack --watch --mode production"
+  },
```
[the remark this answers is not in the document]
Comment (voduchau) [reply demos/typescript-app/package.json:9]: I agree with you, I will update it

File: demos/typescript-app/README.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      value: 'apple',
+    }
+  ],
+  value: '-----',
+  label: 'Fruit',
+  error: 'Error occurred!',
+};
+const dropdown = new Dropdown(dropdownProps);
+root.appendChild(dropdown);
+
```
[the remark this answers is not in the document]
Comment (voduchau) [reply demos/typescript-app/README.md:49]: I will confirm and discuss with you at SSR-2247

[the remark this answers is not in the document]
Comment (TomokoMiyake) [reply demos/typescript-app/README.md:49]: We decided not to publish CustomEventDetail in the next version. We will consider later.

**Review by TomokoMiyake — APPROVED** [review]
LGTM

File: demos/typescript-app/README.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    {
+      label: 'Orange',
+      value: 'orange',
+    },
+    {
+      label: 'Apple',
+      value: 'apple',
+    }
+  ],
```
Comment (TomokoMiyake) [inline demos/typescript-app/README.md:42]: Please add DropdownItem here, too.

**Review by TomokoMiyake — APPROVED** [review]
LGTM!

**Pull request merged**